x/interchainqueries/keeper/keeper.go

```diff
@@ -132,9 +132,51 @@ func (k Keeper) GetAllRegisteredQueries(ctx sdk.Context) []*types.RegisteredQuer
 	return queries
 }
 
-func (k Keeper) RemoveQueryByID(ctx sdk.Context, id uint64) {
+// RemoveQuery removes the given query and relative result data from the store. For a KV query it
+// deletes the *types.QueryResult stored by the query ID, for a TX query it stores the query ID to
+// the list of queries to be removed so the ICQ module can remove the query hashes later.
+func (k Keeper) RemoveQuery(ctx sdk.Context, query *types.RegisteredQuery) {
 	store := ctx.KVStore(k.storeKey)
-	store.Delete(types.GetRegisteredQueryByIDKey(id))
+	store.Delete(types.GetRegisteredQueryByIDKey(query.Id))
+	queryType := types.InterchainQueryType(query.GetQueryType())
+	switch {
+	case queryType.IsKV():
+		store.Delete(types.GetRegisteredQueryResultByIDKey(query.Id))
+	case queryType.IsTX():
+		store.Set(types.GetTxQueryToRemoveByIDKey(query.Id), []byte{})
+	}
+}
+
+// TxQueriesCleanup cleans the module store from obsolete registered TX queries and relative
+// stored transaction hashes. Cleans up to params.TxQueryRemovalLimit hashes at a time or all
+// the hashes if params.TxQueryRemovalLimit is 0.
+func (k Keeper) TxQueriesCleanup(ctx sdk.Context) {
+	rmLimit := k.GetParams(ctx).TxQueryRemovalLimit
+	limited := rmLimit != 0
+
+	queriesToRm := make([]*TxQueryToRemove, 0, rmLimit/10)
```

Comment:
Could you explain why do you divide `rmLimit` by 10 here?

Reply:
just to preallocate some memory. using the whole `rmLimit` would be too much imo because I believe the case when we have `rmLimit` number of queries to be removed is not plausible. on the other hand, leaving it without any preallocation is not efficient